### PR TITLE
add option for hiding header code

### PR DESCRIPTION
**Pull request by ubsan**
I want to see what people's opinions are, on what we should do here. This is _not_ ready to be merged. See #1031 for a little more info.

### CODE REVIEW

File: views/templates.pug [hunk 1]
```diff
@@ -62,6 +62,8 @@
           span 11010
         button.btn.btn-godbolt.btn-sm(title="Execute the binary" data-bind="execute")
           span ./a.out
+        button.btn.btn-godbolt.btn-sm.active.nonbinary(title="Filter functions from headers from the output" data-bind="libraryCode")
+          span .cpp:
```
[the remark this answers is not in the document]
Comment (ubsan) [reply views/templates.pug]: yeah, this is definitely just a prototype :)

**Comment by mattgodbolt** [comment]
I do think I like this feature. What's needed to get it merged? Non-header'd language support? Just dropping the icon on non-C++ languages?

**Comment by partouf** [comment]
It might be useful for other languages, but the mechanism to detect it might be slightly different.
Everything that uses the default asmparser is probably fine to use this if we put the code in asmparser and not just asmparser-vc.

**Comment by mattgodbolt** [comment]
Any thoughts on whether we should merge this, @ubsan ? Seems like a good thing to have in the toolbag, as an option. As @partouf says, we'll need (eventually!) to have hooks for other languages, but this is a great start.

If yo udon't have time to consider it @ubsan I can pick this up! Let me know.

**Comment by partouf** [comment]
Rebased the branch and added filtering to the regular C++ parser.
It retains the external function labels, but leaves out the implementation.

Also renamed the button to "lib.f:" - I don't know if that's any good

**Comment by partouf** [comment]
merging, relevant for Windows VC @Rastaban

**Pull request merged**